dotnet/samples/Concepts/Search/TextSearchPluginExample.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Plugins.Web;
+using Microsoft.SemanticKernel.Plugins.Web.Bing;
+
+namespace Search;
+
+/// <summary>
+/// This example shows how to create and use a <see cref="TextSearchPlugin{T}"/>.
+/// </summary>
+public sealed class TextSearchPluginExample(ITestOutputHelper output) : BaseTest(output)
+{
+    /// <summary>
+    /// Show how to create a <see cref="TextSearchPlugin{T}"/> and use it to perform a text search.
+    /// </summary>
+    [Fact]
+    public async Task SearchAsync()
+    {
+        // Create a search service with Bing search service
+        var searchService = new BingTextSearchService(
+            endpoint: TestConfiguration.Bing.Endpoint,
+            apiKey: TestConfiguration.Bing.ApiKey);
+
+        // Build a kernel with Bing search service and add a text search plugin
+        Kernel kernel = new();
+        var stringPlugin = new TextSearchPlugin<string>(searchService);
+        kernel.ImportPluginFromObject(stringPlugin, "StringSearch");
+        var pagePlugin = new TextSearchPlugin<BingWebPage>(searchService);
+        kernel.ImportPluginFromObject(pagePlugin, "PageSearch");
+
+        // Invoke the plugin to perform a text search and return string values
+        var question = "What is the Semantic Kernel?";
+        var function = kernel.Plugins["StringSearch"]["Search"];
+        var result = await kernel.InvokeAsync(function, new() { ["query"] = question });
+
+        Console.WriteLine(result);
+
+        // Invoke the plugin to perform a text search and return BingWebPage values
+        function = kernel.Plugins["PageSearch"]["Search"];
+        result = await kernel.InvokeAsync(function, new() { ["query"] = question, ["count"] = 2 });
```

Comment:
`SearchAsync` method is generic, and each implementation has these checks: `typeof(T) == typeof(string)` or `typeof(T) == typeof(TextSearchResult)`. `TextSearchPlugin` class is also generic, and for each specific type you need to create separate instance (e.g. `TextSearchPlugin<string>` and `TextSearchPlugin<BingWebPage>`). 

My point during initial discussion was - with generics and type checking it's already a sacrifice, but at the end of the day we still can get only `string` from `kernel.InvokeAsync`, and if it's object, it will be serialized. I think it will be really useful to get the actual search result type like `result.GetValue<BingWebPage>` after `InvokeAsync` method. Otherwise, we can probably avoid using generics, work just with `object`, and the result of this sample will still be the same. 

Just some ideas what we can do in theory:
1. Make `ITextSearchService<T>` generic.
2. For specific service (e.g. `BingTextSearchService`), implement interface with all possible types that can be produced by this service: `BingTextSearchService: ITextSearchService<string>, ITextSearchService<BingWebPage>` - this should allow to avoid `typeof(T) == typeof(string)` check in `SearchAsync` implementation.
3. Based on that, we don't need `TextSearchPlugin` class, because we can mark all `SearchAsync` methods in `BingTextSearchService` as `KernelFunction` and name them as you did in line 27 and 29 - `StringSearch` and `PageSearch`. We will end up with `BingPlugin-StringSearch` and `BingPlugin-PageSearch` in kernel. If we share it with LLM, it will be able to decide which function to use based on user prompt. After specific function is invoked, it will return specific type, which will be possible to get with `result.GetValue<T>`.

Another way for point 3, we can keep `TextSearchPlugin` as non-generic, which will accept `ITextSearchService<string>` and/or `ITextSearchService<WebPage>` in constructor. Then, it will be possible to inject any service that implements `ITextSearchService<string>` and/or `ITextSearchService<WebPage>` (e.g. Bing or Google). In this case, you won't need to initialize `TextSearchPlugin` multiple times, DI can resolve all dependencies for you, there will be no need in type checking in search methods and complex type should be available as result of `InvokeAsync` method.

Let me know what you think about it.